src/devices/bus/vme/vme_hcpu30.cpp

```diff
@@ -55,49 +64,92 @@ DEFINE_DEVICE_TYPE(VME_HCPU30, vme_hcpu30_card_device, "hcpu30", "Besta HCPU30 C
+// shared memory with iocpu
+	map(0x00000000, 0x00000007).w(FUNC(vme_hcpu30_card_device::bootvect_w));
+	map(0x00000000, 0x00000007).r(FUNC(vme_hcpu30_card_device::bootvect_r));
+	map(0x00000008, 0x003fffff).ram().share("dram");
+	map(0x00400000, 0x00ffffff).ram().share("hldram");
+	map(0xffff8000, 0xffff9fff).ram().region("mailbox", 0);
+}
+
+void vme_hcpu30_card_device::cpu_space_map(address_map &map)
+{
+	map(0xfffffff0, 0xffffffff).lr16(NAME([this](offs_t offset) -> u16 { return 0x18 + offset; }));
+}
+
+void vme_hcpu30_card_device::oscpu_space_map(address_map &map)
+{
+	map(0xfffffff0, 0xffffffff).lr16(NAME([this](offs_t offset) -> u16 {
+		logerror("68030 iack %d = %02x\n", offset, m_mailbox[offset ^ 1] >> 8);
+		if (1 || BIT(m_irq_state, 6)) // XXX
+		{
+			m_irq_state &= ~(1 << 6); // raise IRQ30*
+			update_030_irq(offset, CLEAR_LINE);
+			return m_mailbox[offset ^ 1] >> 8;
+		}
+		else
+		{
+			return 0;
+		}
+	}));
 }
 
 static INPUT_PORTS_START(hcpu30)
 	PORT_START("SA1")
-	PORT_DIPNAME(0x03000000, 0x00000000, "Console port speed")
-	PORT_DIPSETTING(0x00000000, "9600")
-	PORT_DIPSETTING(0x01000000, "19200")
-	PORT_DIPSETTING(0x02000000, "38400")
-	PORT_DIPSETTING(0x03000000, "4800")
-	PORT_DIPNAME(0x04000000, 0x04000000, "Boot into...")
-	PORT_DIPSETTING(0x00000000, "UNIX")
-	PORT_DIPSETTING(0x04000000, "Monitor")
-	PORT_DIPNAME(0x10000000, 0x00000000, "VME bus width")
-	PORT_DIPSETTING(0x00000000, "32 bits")
-	PORT_DIPSETTING(0x10000000, "16 bits")
-	PORT_DIPNAME(0x20000000, 0x00000000, "VME bus free")
-	PORT_DIPSETTING(0x00000000, "ROR")
-	PORT_DIPSETTING(0x20000000, "not ROR")
-	PORT_DIPNAME(0x40000000, 0x00000000, "Cache burst mode")
-	PORT_DIPSETTING(0x00000000, "Off")
-	PORT_DIPSETTING(0x40000000, "On")
-	PORT_DIPNAME(0x80000000, 0x00000000, "Undefined")
-	PORT_DIPSETTING(0x00000000, "Off")
-	PORT_DIPSETTING(0x80000000, "On")
+	PORT_DIPNAME(0x03, 0x00, "Console port speed")
+	PORT_DIPSETTING(0x00, "9600")
+	PORT_DIPSETTING(0x01, "19200")
+	PORT_DIPSETTING(0x02, "38400")
+	PORT_DIPSETTING(0x03, "4800")
+	PORT_DIPNAME(0x04, 0x04, "Boot into...")
+	PORT_DIPSETTING(0x00, "UNIX")
+	PORT_DIPSETTING(0x04, "Monitor")
+	PORT_DIPNAME(0x08, 0x00, "Undefined 1")
+	PORT_DIPSETTING(0x00, "Off")
+	PORT_DIPSETTING(0x08, "On")
+	PORT_DIPNAME(0x10, 0x00, "VME bus width")
+	PORT_DIPSETTING(0x00, "32 bits")
+	PORT_DIPSETTING(0x10, "16 bits")
+	PORT_DIPNAME(0x20, 0x00, "VME bus free")
+	PORT_DIPSETTING(0x00, "ROR")
+	PORT_DIPSETTING(0x20, "REC")
+	PORT_DIPNAME(0x40, 0x00, "Cache burst mode")
+	PORT_DIPSETTING(0x00, "Off")
+	PORT_DIPSETTING(0x40, "On")
+	PORT_DIPNAME(0x80, 0x00, "Undefined 2")
+	PORT_DIPSETTING(0x00, "Off")
+	PORT_DIPSETTING(0x80, "On")
 INPUT_PORTS_END
 
 ROM_START(hcpu30)
 	ROM_REGION32_BE(0x8000, "user1", ROMREGION_ERASEFF)
+	// Rev 1.E of 09-NOV-1993
 	ROM_LOAD("hcpu30.27c256.dat", 0x0000, 0x8000, CRC(d24da66e) SHA1(5431b0559b168a995e725b35e1465a0b8ee8aa72))
+
+	ROM_REGION32_BE(0x8000, "mailbox", ROMREGION_ERASEFF)
```

Comment:
@galibert is unhappy with regions being used as RAM in other drivers. A memory share seems more appropriate here.